File: src/main/java/dk/cngroup/wishlist/DatabaseInitializer.java
package dk.cngroup.wishlist;

import dk.cngroup.wishlist.entity.Client;
import dk.cngroup.wishlist.entity.Product;
import dk.cngroup.wishlist.entity.Wishlist;
import dk.cngroup.wishlist.repository.ClientRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.boot.CommandLineRunner;
import org.springframework.stereotype.Component;

import java.util.List;

@Component
@RequiredArgsConstructor
public class DatabaseInitializer implements CommandLineRunner {

    private final ClientRepository clientRepository;

    @Override
    public void run(String... args) {
        Product tieFighter = new Product("TIE Fighter");
        Product deathStar = new Product("Death Star");
        Product starDestroyer = new Product("Star Destroyer");

        Wishlist wishlist = new Wishlist(List.of(tieFighter, deathStar, starDestroyer));

        Client vader = new Client("Darth", "Vader");
        vader.addWishlist(wishlist);
        clientRepository.save(vader);
    }
}

File: src/main/java/dk/cngroup/wishlist/DatabaseInitializer.kt
package dk.cngroup.wishlist

import dk.cngroup.wishlist.entity.Client
import dk.cngroup.wishlist.entity.ClientRepository
import dk.cngroup.wishlist.entity.Product
import dk.cngroup.wishlist.entity.Wishlist
import org.springframework.boot.CommandLineRunner
import org.springframework.stereotype.Component

@Component
class DatabaseInitializer(private val clientRepository: ClientRepository) : CommandLineRunner {
    override fun run(vararg args: String) {
        val tieFighter = Product(code = "TIE Fighter")
        val deathStar = Product(code = "Death Star")
        val starDestroyer = Product(code = "Star Destroyer")
        val wishlist = Wishlist(products = arrayListOf(tieFighter, deathStar, starDestroyer))
        val vader = Client(firstName = "Darth", lastName = "Vader")
        vader.addWishlist(wishlist)
        clientRepository.save(vader)
    }
}